### PR TITLE
Added enable_nonce

### PR DESCRIPTION
**Pull request by kerneldemon**
Hi there!

It would be great to have an option of having a randomly generated value that can be used in script/style tags. This is viable for sites with a lot of legacy code, where it is difficult to move all css/js code into separate files, and adding a nonce attribute everywhere would be a much easier task.

I moved most of the work done in the csp factory method into the bundle extension.
This was done so that it would be easier to introduce new optional dependencies (like the NonceGenerator I added).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -220,6 +220,16 @@ nelmio_security:
...
 
+(Optional) Use *enable_nonce* to add a nonce value to your every request.
+
+```yaml
+nelmio_security:
+    csp:
+        enable_nonce: true
+```
+
+The nonce token that was used in the current request can be retrieved by using the *nelmio_security.nonce_generator* service.
```
Comment (Seldaek) [inline README.md]: It'd be nice to have an example of how this looks especially with the ->getCurrentToken call so people don't have to dig in the code to see which method should be called.

File: ContentSecurityPolicy/NonceGenerator.php [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    {
+        return $this->currentToken;
+    }
+
+    /**
+     * @return string
+     */
+    protected function buildNonce()
+    {
+        $randomBytes = openssl_random_pseudo_bytes($this->numberOfBytes, $isSecureEnough);
```
Comment (Seldaek) [inline ContentSecurityPolicy/NonceGenerator.php]: Why not use Symfony's SecureRandom class for this?

**Comment by kerneldemon** [comment]
@Seldaek could you please take a second look at this pull request again?

**Comment by romainneutron** [comment]
What the status of this PR? @Seldaek are you in favor of this implementation? I've to admit this is a feature I miss :)

**Comment by Seldaek** [comment]
I just don't have time at the moment.. if you wanna review and try and make sure it works and vouch for it then go ahead and merge :)

**Comment by romainneutron** [comment]
@kerneldemon Do you have anytime to rebase this one?

**Comment by kerneldemon** [comment]
Sure, sometime this week.

**Comment by romainneutron** [comment]
Hello @kerneldemon, thanks for the rebase and good work!
I've superseded your PR in #82, adding more functionality and Twig support out of the box, do you have any thoughts?

**Comment by kerneldemon** [comment]
@romainneutron Great work! I love how you added a twig extension as well - wanted to do that myself, but didn't need it for the legacy code base I was working on at the time.

**Comment by kerneldemon** [comment]
I guess I should close in favor of #82 to avoid duplicate pull requests.

**Pull request closed without merging**